traits/trait_types.py

```diff
@@ -1953,19 +1961,51 @@ def __init__(self, *args, **metadata):
                     "when using the 'values' keyword"
                 )
         else:
-            default_value = args[0]
-            if (len(args) == 1) and isinstance(default_value, EnumTypes):
-                args = default_value
-                default_value = enum_default(args)
-            elif (len(args) == 2) and isinstance(args[1], EnumTypes):
-                args = args[1]
+            if len(args) < 1:
+                raise TraitError("Enum trait requires at "
+                                 "least 1 argument.")
+
+            elif len(args) == 1:
+                arg = args[0]
+                if isinstance(arg, EnumTypes):
+                    default_value = next(iter(arg), None)
+                    self.values = tuple(arg)
+
+                # Treat str, bytes and bytearray as discrete units,
+                # and not as a collection.
+                elif isinstance(arg, (str, bytes, bytearray)):
+                    default_value = arg
+                    self.values = {arg}
```

Comment:
Let's make this a tuple; I don't think there's a good reason to use a set here.

Reply:
That's because a single string value should not be split up into individual letters, which would happen if we use `tuple`.

Reply:
I meant that we should create a tuple of length 1 containing only the item `arg`; not that we should split `arg` up into letters. Currently we're creating a set of length 1 containing only the item `arg`. Let's make it a tuple instead.